src/models/preferencesModel.js

```diff
@@ -0,0 +1,111 @@
+const mysql = require('mysql');
+const { Logger } = require("../utils/log4js.js");
+const log = Logger();
+
+const createPreference = async (db, userID, params) => {
+  const sql = 'INSERT INTO UserPreference (UserID, TypeOfAnimal, Breed, Sex, MinAge, MaxAge, More) '
+    + 'VALUES (?, ?, ?, ?, ?, ?, ?)';
+
+  const vals = [
+    userID,
+    JSON.stringify(params.typeOfAnimal),
+    JSON.stringify(params.breed),
+    params.sex,
+    params.minAge,
+    params.maxAge,
+    JSON.stringify(params.more)
+  ];
+  log.debug("Running createPreference sql = " + mysql.format(sql, vals));
+  return new Promise((resolve, reject) => {
+    db.query(sql, vals, (err, res, fields) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+const getPreferencesByUserID = async (db, userID) => {
+  const sql = 'SELECT * FROM UserPreference WHERE UserID = ?';
+  const vals = [userID];
+  log.debug("Running getPreferencesByUserID sql = " + mysql.format(sql, vals));
+  return new Promise((resolve, reject) => {
+    db.query(sql, vals, (err, res, fields) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+const updatePreferences = async (db, userID, params) => {
+  //get original values
+  let original = await getPreferencesByUserID(db, userID);
+  if (original.length == 0) throw new Error('Cannot find user data where userID=' + userID);
+
+  const sql = 'UPDATE UserPreference SET TypeOfAnimal = ?, Breed = ?, Sex = ?, '
+    + 'MinAge = ?, MaxAge = ?, More = ? '
+    + 'WHERE UserID = ?;'
+
+    // Since every input is optional, always change to null if applicable
+  const values = [
+    JSON.stringify(params.typeOfAnimal),
+    JSON.stringify(params.breed),
+    params.sex,
+    params.minAge,
+    params.maxAge,
+    JSON.stringify(params.more),
+    userID
+  ];
+
+  log.debug("Running updatePreferences sql = " + mysql.format(sql, values));
+  return new Promise((resolve, reject) => {
+    db.query(sql, values, (err, res, fields) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+const deletePreferenceByID = async (db, preferenceID) => {
```

Comment:
My brain is not working well :,( 
When do we use `deletePreferenceByID`? I can see that `deletePreferencesByUserID` is used when a user is removed but I can't come up with the case for this.